From 97761fa3b27cc6cfb8638b0a88f5e5b14d2403fa Mon Sep 17 00:00:00 2001
From: Waveshare_Team <support@waveshare.com>
Date: Mon, 27 Jul 2026 21:26:09 +0800
Subject: [PATCH] ARM: dts: overlays: waveshare-panel-v2: Add 6.25-DSI-TOUCH-A

Add an overlay option for the Waveshare 6.25-DSI-TOUCH-A panel and
touchscreen.

The 6_25_inch_a option selects the two-lane panel and the Sitronix
ST7123 touch controller at I2C address 0x55.

Signed-off-by: Waveshare_Team <support@waveshare.com>
---
 arch/arm/boot/dts/overlays/README                            | 1 +
 .../dts/overlays/vc4-kms-dsi-waveshare-panel-v2-overlay.dts  | 5 ++++-
 2 files changed, 5 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -6089,6 +6089,7 @@ Params: 3_4_inch_c              3.4" 800
         4_3_inch_a              4.3" 480x800 2lane
         5_0_inch_a              5.0" 720x1280 2lane
         5_5_inch_a              5.5" 720x1280 2lane
+        6_25_inch_a             6.25" 720x1560 2lane
         7_0_inch_a              7.0" A 720x1280 2lane
         7_0_inch_b              7.0" B 720x1280 2lane
         7_0_inch_c              7.0" C 1024x600 2lane
--- a/arch/arm/boot/dts/overlays/vc4-kms-dsi-waveshare-panel-v2-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-dsi-waveshare-panel-v2-overlay.dts
@@ -25,7 +25,7 @@
 				enable-gpio = <&display_mcu 2 GPIO_ACTIVE_HIGH>;
 			};
 
-			touch: goodix@5d {
+			touch: touch@5d {
 				compatible = "goodix,gt9271";
 				reg = <0x5d>;
 				reset-gpio = <&display_mcu 9 GPIO_ACTIVE_HIGH>;
@@ -84,6 +84,9 @@
 		4_3_inch_a = <&dsi_panel>, "compatible=waveshare,4.3-dsi-touch-a";
 		5_0_inch_a = <&dsi_panel>, "compatible=waveshare,5.0-dsi-touch-a";
 		5_5_inch_a = <&dsi_panel>, "compatible=waveshare,5.5-dsi-touch-a";
+		6_25_inch_a = <&dsi_panel>, "compatible=waveshare,6.25-dsi-touch-a",
+			      <&touch>, "compatible=sitronix,st7123",
+			      <&touch>, "reg:0=0x55";
 		7_0_inch_a = <&dsi_panel>, "compatible=waveshare,7.0-dsi-touch-a";
 		7_0_inch_b = <&dsi_panel>, "compatible=waveshare,7.0-dsi-touch-b";
 		7_0_inch_c = <&dsi_panel>, "compatible=waveshare,7.0-dsi-touch-c";
